### PR TITLE
[CIFuzz] Fix repo detection for repos downloaded through "go get".

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Superseeds: #5463
Fixes: #3670

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
CC @catenacyber

**Review by inferno-chromium — APPROVED** [review]

File: infra/base-images/base-builder/detect_repo.py [hunk 1]
```diff
@@ -107,20 +107,25 @@ def get_repo(repo_path):
...
-                   location=repo_path)
-  out = out.split('/')[-1].replace('.git', '').rstrip()
-  return out == repo_name
+  repo_url, _ = execute(['git', 'config', '--get', 'remote.origin.url'],
+                        location=repo_path)
+  # Handle two common cases:
+  # https://github.com/google/syzkaller/
+  # https://github.com/google/syzkaller.git
+  repo_url = repo_url.replace('.git', '').rstrip('/')
+  actual_repo_name = repo_url.split('/')[-1].rstrip()
```
Comment (catenacyber) [inline infra/base-images/base-builder/detect_repo.py:127]: I think the first step is to `rstrip()` the new line from stdout

**Comment by catenacyber** [comment]
@jonathanmetzman Did you check this with running `python /opt/cifuzz/detect_repo.py --repo_name syzkaller` in syzkaller shell ?

**Comment by jonathanmetzman** [comment]
> @jonathanmetzman Did you check this with running `python /opt/cifuzz/detect_repo.py --repo_name syzkaller` in syzkaller shell ?

Just did.

File: infra/base-images/base-builder/detect_repo.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/base-images/base-builder/detect_repo.py]: Thanks!

**Pull request merged**